src/controllers/users.js

```diff
@@ -11,3 +12,34 @@ exports.getUser = async (req, res, next) => {
   // This is an example of a successful response
   res.json({ success: true, data: user }); */
 };
+exports.updateProduct = async (req, res) => {
+  Product.findByIdAndUpdate(
+    req.params.id,
+    {
+      $set: {
+        title: req.body.title,
+        description: req.body.description,
+        price: req.body.price,
+        category: req.body.category,
+        coverImage: req.body.coverImage,
+        images: req.body.images,
+        type: req.body.type,
+        location: req.body.location,
+        status: req.body.status,
+        condition: req.body.condition,
+      },
+    },
+    { new: true }
+  )
+    .then((product) => {
+      res.status(200).json({
+        message: 'product updated successfully',
+        product,
+      });
+    })
+    .catch(() => {
+      res.status(500).json({
+        message: 'Error occured',
+      });
+    });
```

Comment:
since your middles isAdmin and isLogged are empty, you could delete them.

Reply:
ok deleted them, but i think we need to fill them in the future, there are some routes that requires authorization. Do we have to create middlewares for authorization??